diff --git a/dist/extractor/concatClassName.js b/dist/extractor/concatClassName.js
index 707d78c..1c1fdb5 100644
--- a/dist/extractor/concatClassName.js
+++ b/dist/extractor/concatClassName.js
@@ -18,7 +18,7 @@ __export(concatClassName_exports, {
 });
 module.exports = __toCommonJS(concatClassName_exports);
 function concatClassName(_cn) {
-  const args = arguments, usedPrefixes = /* @__PURE__ */ new Set();
+  const args = arguments, usedPrefixes = [];
   let final = "";
   const len = args.length;
   let propObjects = null;
@@ -42,10 +42,10 @@ function concatClassName(_cn) {
         final = name + " " + final;
         continue;
       }
-      const isMediaQuery = name[splitIndex + 1] === "_", styleKey = name.slice(1, name.indexOf("-")), mediaKey = isMediaQuery ? name.slice(splitIndex + 2, splitIndex + 7) : null, uid = mediaKey ? styleKey + mediaKey : styleKey;
-      if (usedPrefixes.has(uid))
+      const isMediaQuery = name[splitIndex + 1] === "_", styleKey = name.slice(1, name.lastIndexOf("-")), mediaKey = isMediaQuery ? name.slice(splitIndex + 2, splitIndex + 7) : null, uid = mediaKey ? styleKey + mediaKey : styleKey;
+      if (usedPrefixes.indexOf(uid) > -1)
         continue;
-      usedPrefixes.add(uid);
+      usedPrefixes.push(uid);
       const propName = styleKey;
       propName && propObjects && propObjects.some((po) => {
         if (mediaKey) {
@@ -56,7 +56,7 @@ function concatClassName(_cn) {
       }) || (final = name + " " + final);
     }
   }
-  return final.trim();
+  return final;
 }
 const pseudoInvert = {
   hover: "hoverStyle",
diff --git a/dist/extractor/concatClassName.mjs b/dist/extractor/concatClassName.mjs
index c071801..df0127e 100644
--- a/dist/extractor/concatClassName.mjs
+++ b/dist/extractor/concatClassName.mjs
@@ -25,7 +25,7 @@ __export(concatClassName_exports, {
 module.exports = __toCommonJS(concatClassName_exports);
 function concatClassName(_cn) {
   const args = arguments,
-    usedPrefixes = /* @__PURE__ */new Set();
+    usedPrefixes = [];
   let final = "";
   const len = args.length;
   let propObjects = null;
@@ -51,11 +51,11 @@ function concatClassName(_cn) {
         continue;
       }
       const isMediaQuery = name[splitIndex + 1] === "_",
-        styleKey = name.slice(1, name.indexOf("-")),
+        styleKey = name.slice(1, name.lastIndexOf("-")),
         mediaKey = isMediaQuery ? name.slice(splitIndex + 2, splitIndex + 7) : null,
         uid = mediaKey ? styleKey + mediaKey : styleKey;
-      if (usedPrefixes.has(uid)) continue;
-      usedPrefixes.add(uid);
+      if (usedPrefixes.indexOf(uid) > -1) continue;
+      usedPrefixes.push(uid);
       const propName = styleKey;
       propName && propObjects && propObjects.some(po => {
         if (mediaKey) {
@@ -66,7 +66,7 @@ function concatClassName(_cn) {
       }) || (final = name + " " + final);
     }
   }
-  return final.trim();
+  return final;
 }
 const pseudoInvert = {
   hover: "hoverStyle",
diff --git a/dist/extractor/extractToClassNames.js b/dist/extractor/extractToClassNames.js
index 5c4b6f5..3808fe7 100644
--- a/dist/extractor/extractToClassNames.js
+++ b/dist/extractor/extractToClassNames.js
@@ -38,7 +38,7 @@ async function extractToClassNames({
   const tm = (0, import_timer.timer)(), { getCSSStylesAtomic, createMediaStyle } = (0, import_requireTamaguiCore.requireTamaguiCore)("web");
   if (sourcePath.includes("node_modules"))
     return null;
-  if (shouldPrintDebug && console.warn(`--- ${sourcePath} --- 
+  if (shouldPrintDebug && console.warn(`--- ${sourcePath} ---
 
 `), typeof source != "string")
     throw new Error("`source` must be a string of javascript");
@@ -243,7 +243,9 @@ async function extractToClassNames({
           if (!ternary.consequent) continue;
           const classNames = addStyles(ternary.consequent);
           ternary.fontFamily && classNames.unshift(`font_${ternary.fontFamily}`);
-          const baseString = t.isStringLiteral(baseClassNameExpression) ? baseClassNameExpression.value : "", fullClassNameWithDups = (baseString ? `${baseString} ` : "") + classNames.join(" "), fullClassName = (0, import_concatClassName.concatClassName)(fullClassNameWithDups), classNameLiteral = t.stringLiteral(fullClassName);
+          const baseString = t.isStringLiteral(baseClassNameExpression) ? baseClassNameExpression.value : "", fullClassName = (0, import_concatClassName.concatClassName)(
+            (baseString ? `${baseString} ` : "") + classNames.join(" ")
+          ), classNameLiteral = t.stringLiteral(fullClassName);
           ternaryClassNameExpr ? ternaryClassNameExpr = t.conditionalExpression(
             ternary.test,
             classNameLiteral,
@@ -283,14 +285,14 @@ async function extractToClassNames({
   );
   return shouldPrintDebug && (console.info(
     `
- -------- output code ------- 
+ -------- output code -------
 
 `,
     result.code.split(`
 `).filter((x) => !x.startsWith("//")).join(`
 `)
   ), console.info(`
- -------- output style -------- 
+ -------- output style --------
 
 `, styles)), printLog(res), {
     ast,
diff --git a/dist/extractor/extractToClassNames.mjs b/dist/extractor/extractToClassNames.mjs
index e07ac40..ef04a03 100644
--- a/dist/extractor/extractToClassNames.mjs
+++ b/dist/extractor/extractToClassNames.mjs
@@ -63,7 +63,7 @@ async function extractToClassNames({
       createMediaStyle
     } = (0, import_requireTamaguiCore.requireTamaguiCore)("web");
   if (sourcePath.includes("node_modules")) return null;
-  if (shouldPrintDebug && console.warn(`--- ${sourcePath} --- 
+  if (shouldPrintDebug && console.warn(`--- ${sourcePath} ---
 
 `), typeof source != "string") throw new Error("`source` must be a string of javascript");
   if (!path.isAbsolute(sourcePath)) throw new Error("`sourcePath` must be an absolute path to a .js file, got: " + sourcePath);
@@ -232,8 +232,7 @@ async function extractToClassNames({
           const classNames = addStyles(ternary.consequent);
           ternary.fontFamily && classNames.unshift(`font_${ternary.fontFamily}`);
           const baseString = t.isStringLiteral(baseClassNameExpression) ? baseClassNameExpression.value : "",
-            fullClassNameWithDups = (baseString ? `${baseString} ` : "") + classNames.join(" "),
-            fullClassName = (0, import_concatClassName.concatClassName)(fullClassNameWithDups),
+            fullClassName = (0, import_concatClassName.concatClassName)((baseString ? `${baseString} ` : "") + classNames.join(" ")),
             classNameLiteral = t.stringLiteral(fullClassName);
           ternaryClassNameExpr ? ternaryClassNameExpr = t.conditionalExpression(ternary.test, classNameLiteral, ternaryClassNameExpr) : ternaryClassNameExpr = t.conditionalExpression(ternary.test, classNameLiteral, baseClassNameExpression);
         }
@@ -261,12 +260,12 @@ async function extractToClassNames({
       sourceMaps: !0
     }, source);
   return shouldPrintDebug && (console.info(`
- -------- output code ------- 
+ -------- output code -------
 
 `, result.code.split(`
 `).filter(x => !x.startsWith("//")).join(`
 `)), console.info(`
- -------- output style -------- 
+ -------- output style --------
 
 `, styles)), printLog(res), {
     ast,
diff --git a/dist/registerRequire.js b/dist/registerRequire.js
index 7aee7ed..3ced1d9 100644
--- a/dist/registerRequire.js
+++ b/dist/registerRequire.js
@@ -56,12 +56,14 @@ function registerRequire(platform, { proxyWormImports } = {
       return og.apply(this, ["react-native-web/package.json"]);
     if (path === "@tamagui/react-native-web-lite" || path === "react-native" || path.startsWith("react-native/"))
       try {
-        return og.apply("react-native");
+        return og.apply(this, ["react-native"]);
       } catch {
         return og.apply(this, ["@tamagui/react-native-web-lite"]);
       }
     if (!whitelisted[path] && proxyWormImports && !path.includes(".tamagui-dynamic-eval"))
       return path === "tamagui" ? og.apply(this, [path]) : proxyWorm;
+    if (path === "punycode")
+      return og.apply(this, ["punycode/"]);
     try {
       return og.apply(this, arguments);
     } catch (err) {
diff --git a/dist/registerRequire.mjs b/dist/registerRequire.mjs
index d87ab8d..3f7fd6b 100644
--- a/dist/registerRequire.mjs
+++ b/dist/registerRequire.mjs
@@ -64,11 +64,12 @@ function registerRequire(platform, {
     if (path === "react-native-svg") return og.apply(this, ["@tamagui/react-native-svg"]);
     if (path === "react-native/package.json") return og.apply(this, ["react-native-web/package.json"]);
     if (path === "@tamagui/react-native-web-lite" || path === "react-native" || path.startsWith("react-native/")) try {
-      return og.apply("react-native");
+      return og.apply(this, ["react-native"]);
     } catch {
       return og.apply(this, ["@tamagui/react-native-web-lite"]);
     }
     if (!whitelisted[path] && proxyWormImports && !path.includes(".tamagui-dynamic-eval")) return path === "tamagui" ? og.apply(this, [path]) : proxyWorm;
+    if (path === "punycode") return og.apply(this, ["punycode/"]);
     try {
       return og.apply(this, arguments);
     } catch (err) {
